[richiehx] iP

richiehx opened the pull request:
# Nano the Chatbot
> "If ur asking, u want it" - Anonymous

Nano frees your mind of having to remember things you need to do. It's,
- Text-based
- Easy to learn
- Killing me

All you need to do is,
1. download it from [here](https://github.com/richiehx/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.
4. Try not to cry. :droplet:
5. ???
6. Profit :+1:

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---
If you Java programmer, you can use it to practice Java too. Here's the main method:
```
public static void main(String[] args) {
        new Duke().run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

richiehx committed:
Completed Level-0

richiehx committed:
Completed Level-1

richiehx committed:
Completed Level-2

richiehx committed:
Completed Level 3

richiehx committed:
Completed Level 4

richiehx committed:
Completed Automated Text UI Testing

richiehx committed:
Completed Level 5

richiehx committed:
Completed Level 6

richiehx committed:
Fixed bug when entering strings as index for mark/unmark and delete commands

richiehx committed:
Refactored code to use Enums

richiehx committed:
Changed test files due to change of output

richiehx committed:
Add File Save and Read functionality

richiehx committed:
Complete Level 7 and Refactor Code into TaskList and Storage

richiehx committed:
Merge branch 'branch-Level-7'

richiehx committed:
Refactor Code into Parser and TaskList

richiehx committed:
Complete Level 8 and Add Date parsing

richiehx committed:
Merge branch 'branch-Level-8'

richiehx committed:
Refactor Code and Added Ui class for more OOP

richiehx committed:
Merge branch 'branch-A-MoreOOP'

richiehx committed:
Refactor files into packages

richiehx committed:
Merge branch 'branch-A-Packages'

richiehx committed:
Fix bug from Storage Constructor

richiehx committed:
Merge branch 'branch-A-Packages'

richiehx committed:
Merge branch 'add-gradle-support' of https://github.com/richiehx/ip

richiehx committed:
Use Gradle to build and run and Fix Text for tasks

richiehx committed:
Merge branch 'branch-A-Gradle'

richiehx committed:
Add JUnit tests for Parser and TaskList

richiehx committed:
Merge branch 'branch-A-JUnit'

richiehx committed:
Complete A-Jar

richiehx committed:
Merge branch 'branch-A-Jar'

richiehx committed:
Add JavaDocs

richiehx committed:
Modify files for Coding Standard

richiehx committed:
Add Find Functionality

richiehx committed:
Merge branch 'branch-A-JavaDoc'

richiehx committed:
Merge branch 'branch-A-CodingStandard'

richiehx committed:
Merge branch 'branch-Level-9'

richiehx committed:
Add Files for the Gui

LWZ19 commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,54 @@
+    TODO("todo"),
+    DEADLINE("deadline"),
+    EVENT("event"),
+    FIND("find");
```
Nice use of enums!! 😄

LWZ19 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,115 @@
+                    case LIST:
+                        ui.printOutput(tasks.formatList());
+                        break;
+                    case MARK:
```
Perhaps you could include a `//fallthrough` comment, for example
```suggestion
                    case MARK: //fallthrough
```

LWZ19 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,115 @@
+                        ui.printOutput("I don't understand what you're saying.");
+                        break;
+                }
+            } catch (DukeException e) {
```
Good job for using switch statements, which makes your code clean and easy to understand!
But, perhaps you could abstract out the handling of the different commands to another class?

LWZ19 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,115 @@
+    /** The UI to interact with the User. */
+    private Ui ui;
+
+    /** The file path to store the User's Task Data. */
```
I like how you took the effect of writing JavaDoc comments for each attribute and method! :thumbsup:

LWZ19 commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,54 @@
+    /**
+     * The String value of the command.
+     */
+    private String value;
```
Perhaps a more intuitive variable name here? So that it is consistent with the getter method.

LWZ19 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,151 @@
+
+        return new String[]{desc, start, end};
+    }
+
```
Great work, It's nice to see you have different parse methods for different commands!

LWZ19 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,151 @@
+        if (startEnd.length != 2) {
+            throw new DukeException("Hey, the Event given is Invalid!" +
+                    " Make sure that you follow this format:\n" +
+                    Ui.INDENTATION + " 'eventDescription /from startTime /to endTime'");
```
Perhaps you could abstract out different exceptions tailored to different problems? So that you can reduce the amount of reputation in the code.

LWZ19 commented on src/test/java/duke/parser/ParserTest.java:
```diff
@@ -0,0 +1,82 @@
+import duke.exception.DukeException;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
```
It would be better to import everything you need, rather than using the wildcard('`*`')

LWZ19 commented on src/main/java/duke/tasks/TaskList.java:
```diff
@@ -0,0 +1,200 @@
+ */
+public class TaskList {
+    /** The Indentation Level to format text. */
+    private static final String INDENTATION = Ui.INDENTATION;
```
Any reason why you did it this way instead of using `Ui.INDENTATION` directly?

LWZ19 commented on src/main/java/duke/tasks/TaskList.java:
```diff
@@ -0,0 +1,200 @@
+            Task selectedTask = taskList.get(index);
+            if (status.equals("mark")) {
+                selectedTask.markTask();
+                buffer = "Nice! I've marked the task as done:\n   " +
+                        INDENTATION + selectedTask;
```
Perhaps you could add the line break before the `+` operator.
```suggestion
                buffer = "Nice! I've marked the task as done:\n   "
                         + INDENTATION + selectedTask;
```

LWZ19 commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,68 @@
+    public void printOutput(String rawOutput) {
+        System.out.println(formatOutput(rawOutput));
+    }
+}
```
Nice work on the text UI! :blush: Perhaps you could include more features in the Ui, for example, Ui can be used to handle most of the display interactions with the user.

LWZ19 commented on src/test/java/duke/parser/ParserTest.java:
```diff
@@ -0,0 +1,82 @@
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class ParserTest {
```
Nice to see you have written some unit tests!

LWZ19 reviewed (commented):
LGTM! Some suggestions are your UI class could be used to handle more user interactions and just some small nits on the line break before an operator.

Overall, you have done an excellent job and is on par with code qualities and standard. Very intuitive naming for methods and variables. Keep it up! 🎉

richiehx committed:
Complete Level 10 of the Chatbot

richiehx committed:
Merge branch 'branch-Level-10'

richiehx committed:
Add checkstyle and Fix violations

richiehx committed:
Merge branch 'branch-A-CheckStyle'

seanpzk commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,109 @@
+    public static void main(String[] args) {
+        new Duke().run();
+    }
+}
```
I like how you packaged your different classes!

seanpzk commented on src/main/java/duke/tasks/Event.java:
```diff
@@ -0,0 +1,75 @@
+    public String toString() {
+        return "[E]" + super.toString() + " (from: " + this.start + " to: " + this.end + ")";
+    }
+}
```
JavaDocs are very clear and concise :)

seanpzk commented on src/test/java/duke/parser/ParserTest.java:
```diff
@@ -0,0 +1,81 @@
+        assertEquals(expected, exception.getMessage());
+    }
+
+}
```
I like how the test names are clear and easy to know what you are trying to test for. Also good effort on implementing the different test cases to test both positive and negative test cases

seanpzk commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,148 @@
+
+        return new String[]{desc, start, end};
+    }
+
```
Good job, the JavaDoc comments follow the coding standard closely and is clear in explaining the purposes of the functions and variables!

richiehx committed:
Fix files for building

richiehx committed:
Fix files for building

Gradle is not building the files correctly due to the wrong
mainClass set and the checkstyle errors being thrown.

Modifications are required to allow Gradle to build the files
properly and bundle them into a JAR file.

Let's,
* modify the build.gradle file to use the correct mainClass.
* update the test files to conform to the checkstyle.

richiehx committed:
Merge branch 'master' of https://github.com/richiehx/ip

richiehx committed:
Implement Assertions

No assertions to check for Developer errors in code.

Assertions help to define assumptions about the program state so
that the runtime can verify them.

Let's add assertions to detect possible bugs in the code.

richiehx committed:
Refactor files to Improve Code Quality

Files do not have the suggested Code Quality.

Better Code Quality will help with Readability of the Code.

Let's refactor the files to follow the suggested Code Quality
improvements.

richiehx committed:
Set up Continuous Integration

Use GitHub Actions workflow to set up Continuous Integration.

To automate integration of code changes into the project.

Let's set up the workflow .yml file.

richiehx committed:
Fix workflow .yml file

richiehx committed:
Merge pull request #1 from richiehx/branch-A-Assertions

Implement Assertions

richiehx committed:
Merge branch 'master' into branch-A-CodeQuality

richiehx committed:
Merge pull request #2 from richiehx/branch-A-CodeQuality

Refactor files to Improve Code Quality

richiehx committed:
Merge pull request #3 from richiehx/branch-A-CI

Set up Continuous Integration

richiehx committed:
Add Edit Functionality

Edit command to edit tasks in Task List.

No method to edit tasks in Task List.

Let's add an Edit method to Edit tasks.

richiehx committed:
Fix code to conform to checkstyle

richiehx committed:
Merge pull request #4 from richiehx/branch-C-Update

Add Edit Functionality

richiehx committed:
Improve Nano User and Bot Images

richiehx committed:
Change Ui image

richiehx committed:
Merge pull request #6 from richiehx/branch-A-BetterGui

Improve Nano User and Bot Images

richiehx committed:
Update README.md

richiehx committed:
Edit User Guide

richiehx committed:
Fix Tests from modifying display message

richiehx committed:
Fix typo

richiehx committed:
Merge pull request #7 from richiehx/branch-A-UserGuide

Edit User Guide

richiehx committed:
Update README.md

richiehx committed:
Update README.md

richiehx committed:
Refactor code according to SLAP

richiehx committed:
Merge pull request #8 from richiehx/branch-RefactorCode

Refactor code according to SLAP

richiehx committed:
Update Download Link

richiehx committed:
Fix JavaFX version in fxml files

richiehx committed:
Merge pull request #9 from richiehx/branch-JavaFxFix

Fix JavaFX version in fxml files
